### PR TITLE
[JENKINS-29910] Add support for Ivy plugin

### PR DESCRIPTION
**Pull request by tbingaman**

### CODE REVIEW

**Comment by daspilker** [comment]
Looks good so far. Please add a some tests for the builders, e.g. a `IvyBuilderContextSpec` and add some tests to `JobParentSpec` for the `ivyJob` method.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/JobParent.groovy [hunk 1]
```diff
@@ -51,6 +52,15 @@ abstract class JobParent extends Script implements DslFactory {
     }
 
     /**
+     * @since 1.38
+     */
+    @Override
+    @RequiresPlugin(id = 'ivy')
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/JobParent.groovy]: add a minimum version, e.g. the latest version 1.24.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/jobs/IvyBuilderContext.groovy [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    @Override
+    protected void addExtensionNode(Node node) {
+        ivyBuilderNodes << node
+    }
+
+    void ant(@DslContext(AntContext) Closure antClosure = null) {
+        ant(null, null, null, antClosure)
+    }
+
+    void ant(String targetsStr, @DslContext(AntContext) Closure antClosure = null) {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/jobs/IvyBuilderContext.groovy]: Remove the overloaded `ant` methods, the Closure-only variant is enough. Configuring options as method parameters and context methods is confusing and makes the implementation complicate. The overloaded methods have not been removed from `StepContext` to maintain compatibility, but they to not need to be replicated by new contexts.

**Comment by tbingaman** [comment]
All suggested changes made, squashed and rebased on latest master.

**Comment by daspilker** [comment]
Thanks for contributing a new job type!

**Pull request merged**